### PR TITLE
fix: alter procedure table not found issue

### LINKED ISSUE #1992: Failed to retry alter procedure
### What type of bug is this?

Unexpected error

### What subsystems are affected?

Meta

### What happened?

2023-07-18T16:48:10.562049Z ERROR client::database: Failed to do Flight get, addr: 127.0.0.1:13001, code: Internal error, source: Failed to do Flight get, code: Internal error, source: status: Internal, message: "Failed to wait procedure done, source: Failed to execute procedure, source: Procedure retry exceeded max times, procedure_id: 7346b26b-2344-4617-98f8-888c24775377, source:Procedure exec failed, source: Expected to retry later, reason: Failed to execute operation on datanode, source: Failed to do Flight get, code: Some requested entity was not found, source: Table not found: greptime.public.my_table_rename", details: [], metadata: MetadataMap { headers: {"content-type": "application/grpc", "date": "Tue, 18 Jul 2023 16:48:10 GMT", "content-length": "0"} }
thread 'tokio-runtime-worker' panicked at 'assertion failed: `(left == right)`
  left: `"Affected Rows: 0"`,
 right: `"Error: 1003(Internal), status: Internal, message: \"Failed to wait procedure done, source: Failed to execute procedure, source: Procedure retry exceeded max times, procedure_id: 7346b26b-2344-4617-98f8-888c24775377, source:Procedure exec failed, source: Expected to retry later, reason: Failed to execute operation on datanode, source: Failed to do Flight get, code: Some requested entity was not found, source: Table not found: greptime.public.my_table_rename\", details: [], metadata: MetadataMap { headers: {\"content-type\": \"application/grpc\", \"date\": \"Tue, 18 Jul 2023 16:48:10 GMT\", \"content-length\": \"0\"} }"`', src/chaos/src/tests/alter_table_with_network_delay.rs:154:9

### What operating system did you use?

Ubuntu 23.04

### Relevant log output and stack trace

```bash
2023-07-18T16:48:10.562049Z ERROR client::database: Failed to do Flight get, addr: 127.0.0.1:13001, code: Internal error, source: Failed to do Flight get, code: Internal error, source: status: Internal, message: "Failed to wait procedure done, source: Failed to execute procedure, source: Procedure retry exceeded max times, procedure_id: 7346b26b-2344-4617-98f8-888c24775377, source:Procedure exec failed, source: Expected to retry later, reason: Failed to execute operation on datanode, source: Failed to do Flight get, code: Some requested entity was not found, source: Table not found: greptime.public.my_table_rename", details: [], metadata: MetadataMap { headers: {"content-type": "application/grpc", "date": "Tue, 18 Jul 2023 16:48:10 GMT", "content-length": "0"} }
```

### How can we reproduce the bug?

Injects network delay between the datanode and metasrv, and lets the metasrv retry the alter procedure.

### PR DESCRIPTION
**Pull request by WenyXu**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Fix table not found issue while alter procedure (rename) retrying.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
fix #1992

### CODE REVIEW

File: src/meta-srv/src/procedure/alter_table.rs [hunk 1]
```diff
@@ -95,10 +97,12 @@ impl AlterTableProcedure {
             let expr = self.data.task.alter_table.clone();
             joins.push(common_runtime::spawn_bg(async move {
                 debug!("Sending {:?} to {:?}", expr, client);
-                client
-                    .alter(expr)
-                    .await
-                    .map_err(handle_request_datanode_error(datanode))
+                if let Err(err) = client.alter(expr).await {
+                    if err.status_code() != StatusCode::TableNotFound {
```
Comment (killme2008) [inline src/meta-srv/src/procedure/alter_table.rs:101]: Please add some comments to describe this special processing.

**Comment by WenyXu** [comment]
@killme2008 PTAL

**Review by fengjiachun — APPROVED** [review]

File: src/meta-srv/src/procedure/alter_table.rs [hunk 2]
```diff
@@ -95,10 +97,13 @@ impl AlterTableProcedure {
             let expr = self.data.task.alter_table.clone();
             joins.push(common_runtime::spawn_bg(async move {
                 debug!("Sending {:?} to {:?}", expr, client);
-                client
-                    .alter(expr)
-                    .await
-                    .map_err(handle_request_datanode_error(datanode))
+                if let Err(err) = client.alter(expr).await {
```
Comment (MichaelScofield) [inline src/meta-srv/src/procedure/alter_table.rs:100]: Shouldn't the datanode able to handle the rename alter expr, too?

**Comment by WenyXu** [comment]
@MichaelScofield PTAL

**Review by MichaelScofield — APPROVED** [review]

**Pull request merged**